pkgs/development/java-modules/jnr-posix/default.nix

```diff
@@ -0,0 +1,67 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, dpkg
+, jdk
+, makeWrapper
+, maven
+, which
+}:
+let
+  pname = "jnr-posix";
+  version = "3.1.16";
+
+  src = fetchFromGitHub {
+    owner = "jnr";
+    repo = "jnr-posix";
+    rev = "jnr-posix-${version}";
+    hash = "sha512-NI2u2H5dxp+spyZbbzJeiV3FS38TMDmDqjjZzllZVp2qsbKkvv9+szeobh9Zb7dsIfRxhAQgZvw8rjNXbGmzSw==";
+  };
+
+  deps = stdenv.mkDerivation {
+    name = "${pname}-${version}-deps";
+    inherit src;
+
+    nativeBuildInputs = [ jdk maven ];
+
+    buildPhase = ''
+      mvn package -Dmaven.test.skip=true -Dmaven.repo.local=$out/.m2 -Dmaven.wagon.rto=5000
+    '';
+
+    # keep only *.{pom,jar,sha1,nbm} and delete all ephemeral files with lastModified timestamps inside
+    installPhase = ''
+      find $out/.m2 -type f -regex '.+\(\.lastUpdated\|resolver-status\.properties\|_remote\.repositories\)' -delete
+      find $out/.m2 -type f -iname '*.pom' -exec sed -i -e 's/\r\+$//' {} \;
+    '';
+
+    outputHashAlgo = "sha256";
+    outputHashMode = "recursive";
+    outputHash = "sha256-9PxDQAMPr9gblkv+Zq04vydOE3bAyo8CQy5bLHhiCOk=";
+
+    doCheck = false;
+  };
+in
+stdenv.mkDerivation rec {
+  inherit version pname src;
+
+  nativeBuildInputs = [ jdk maven which ];
+
+  patchPhase = ''
+    sed -i "s/\/usr\/bin\/id/$(which id | sed 's#/#\\/#g')/g" src/main/java/jnr/posix/JavaPOSIX.java
```

Comment:
Overriding `patchPhase` breaks `patches` field. Use `prePatch` or `postPatch` instead. See <https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/patch-phase.md> for more info.

Reply:
For modifying file contents always postPatch should be used